client/src/api/xatabaseFetcher.ts

```diff
@@ -0,0 +1,107 @@
+/* eslint-disable @typescript-eslint/no-throw-literal */
+/* eslint-disable @typescript-eslint/ban-types */
+import type { SimpleError } from './xatabaseResponses';
+
+const resolveUrl = (
+  url: string,
+  _queryParams: Record<string, unknown> = {},
```

Comment:
[Just pointing the obvious to not forget] We need a `qs` implem here!

Reply:
Yeah, I thought of that. We should build our own query string serializer as we aim to be 0-dependency. I would've imported ``qs`` library otherwise.

Reply:
https://nodejs.org/dist/latest-v16.x/docs/api/url.html#class-urlsearchparams

Also available in nodejs API